pandas/core/arrays/arrow/dtype.py

```diff
@@ -93,6 +93,9 @@ def construct_from_string(cls, string: str) -> ArrowDtype:
             )
         if not string.endswith("[pyarrow]"):
             raise TypeError(f"'{string}' must end with '[pyarrow]'")
+        if string == "string[pyarrow]":
+            # Ensure Registry.find skips ArrowDtype to use StringDtype instead
+            raise TypeError("string[pyarrow] should be constructed by StringDtype")
```

Comment:
@jreback @jorisvandenbossche if I move the `str` methods over to `ArrowExtensionArray` should we deprecate `StringArrowArray/StringDtype(storage="pyarrow")`?